fili-core/src/main/java/com/yahoo/bard/webservice/web/ApiFilter.java

```diff
@@ -105,7 +126,22 @@ public ApiFilter withOperation(FilterOperation operation) {
      * @param values The new set of values to use in the new ApiFilter.
      * @return the new ApiFilter object.
      */
-    public ApiFilter withValues(Set<String> values) {
+    public ApiFilter withValues(LinkedHashSet<String> values) {
+        return new ApiFilter(dimension, dimensionField, operation, values);
+    }
+
+    /**
+     * Creates a new ApiFilter based on this ApiFilter, except with the new set of filter values replacing this
+     * ApiFilter's filter values.
+     *
+     * @param values The new set of values to use in the new ApiFilter.
+     * @return the new ApiFilter object.
+     *
+     * @deprecated the order of insertion of filter values to be preserved, use
+     * {@link #withValues(LinkedHashSet)} method instead
+     */
+    @Deprecated
+    public ApiFilter withValues(Collection<String> values) {
```

Comment:
[ditto](https://github.com/yahoo/fili/pull/807/files#diff-da59bb50f2ea4852c2fc8cadad5b7be6R58)

This seems to be the time when we must violate the guidelines. In this case, we should make sure the two behave exactly the same. One way to do this is to have `withValues(Collection<String>)` call `withValues(LinkedHashSet<String>)`

Reply:
Refactored the code